GameDataParser/Parsers/MapEntityParser.cs

```diff
@@ -100,386 +68,345 @@ protected override List<MapEntityMetadata> Parse()
-                        else if (modelData.ContainsKey("mixinMS2CubeProp"))
+                        // Only MS2MapProperties has ObjectWeaponItemCode
+                        if (entity is not IMS2MapProperties mapProperties)
                         {
-                            if (!modelData.ContainsKey("ObjectWeaponItemCode"))
-                            {
-                                continue;
-                            }
-                            string weaponId = modelData["ObjectWeaponItemCode"] ?? "0";
-                            if (!weaponId.Equals("0"))
-                            {
-                                // Extract the coordinate from the name. rhy tried just grabbing position, but it wasn't reliable.
-                                Match coordMatch = Regex.Match(name, @"[\-]?\d+[,]\s[\-]?\d+[,]\s[\-]?\d+");
-                                if (coordMatch.Success)
-                                {
-                                    metadata.Objects.Add(new MapObject(CoordB.Parse(coordMatch.Value, ", "), int.Parse(weaponId)));
-                                }
-                            }
+                            continue;
                         }
-                        else if (modelData.ContainsKey("mixinMS2Breakable"))
+
+                        try
                         {
-                            // "mixinMS2Breakable"  But not "mixinMS2BreakableActor", as in ke_fi_prop_buoy_A01_ or el_move_woodbox_B04_
+                            CoordB coord = CoordB.Parse(coordMatch.Value, ", ");
+                            metadata.Objects.Add(new MapObject(coord, int.Parse(mapProperties.ObjectWeaponItemCode)));
                         }
-                        else if (modelData.ContainsKey("mixinMS2RegionBoxSpawn"))
+                        catch (FormatException)
                         {
-                            // "QR_10000264_" is Quest Reward Chest? This is tied to a MS2TriggerAgent making this object appear.
+                            // ignored
+                            //  Console.WriteLine($"Format error parsing {mapProperties.ObjectWeaponItemCode} as int");
+                            //if (mobSpawnData != null)
+                            //{
+                            //    foreach (string s in mobSpawnData.Tags)
+                            //    {
+                            //        byte[] bytes = System.Text.Encoding.UTF8.GetBytes(s);
+                            //        Console.WriteLine($"{Convert.ToHexString(bytes)}");
+                            //    }
+
+                            //    Console.WriteLine($"{mobSpawnData.Difficulty} {mobSpawnData.MinDifficulty}");
+                            //    byte[] bytes = System.Text.Encoding.UTF8.GetBytes(mobSpawnData.Tags[0]);
+                            //    Console.WriteLine($"{Convert.ToHexString(bytes)}, {mobSpawnData.Population}");
+                            //}
                         }
```

Comment:
debugging code?

Reply:
yea, removed partialy, I'll leave in the byte conversion as it will be handy for future debugging.